### ISSUE TITLE
Import from a local json file

### ISSUE THREAD

**Issue by dimidd** [apache/incubator-ponymail#447]
Hello,

Looks like `import-mbox.py` is already using  json as an intermediate format to store messages, before uploading to elasticsearch. Is it possible to import from a json directly? If so, in which format should the json file be?

**Comment by dimidd** [comment]
I've thought about something like this, but not sure how to set the list id.

```import sys
import os
import configparser
import argparse
import json
from elasticsearch import Elasticsearch, helpers


# Fetch config
path = os.path.dirname(os.path.realpath(__file__))
config = configparser.RawConfigParser()
config.read("%s/ponymail.cfg" % path)
auth = None
if config.has_option('elasticsearch', 'user'):
    auth = (config.get('elasticsearch','user'), config.get('elasticsearch','password'))
dbname = config.get("elasticsearch", "dbname")
ssl = config.get("elasticsearch", "ssl", fallback = "False").lower() == 'true'
uri = config.get("elasticsearch", "uri", fallback = "")

es = Elasticsearch(
        [{
            'host': config.get("elasticsearch", "hostname"),
            'port': int(config.get("elasticsearch", "port")),
            'use_ssl': ssl,
            'url_prefix': uri,
            'http_auth': auth
         }],
        max_retries=5,
        retry_on_timeout=True
    )
json_path = sys.argv[1]
with open(json_path, "r") as json_file:
    json = json.load(json_file)
    helpers.bulk(es, json)
```

**Comment by sebbASF** [comment]
ElasticSearch (ES) requires JSON for its API.
The JSON format is based on the requirements of ES and the database schema.
Both may change.

import-mbox already supports various standard mailbox formats (mbox, IMAP etc)

Personally, I don't see a use case for defining another mail input format.

**Comment by dimidd** [comment]
Nothing, I just don't know what should be the exact format of the JSON.
E.g. how to set the list-id that the emails belong to.

2018-05-14 13:22 GMT+03:00 Daniel Gruno <notifications@github.com>:

**Comment by sebbASF** [comment]
As I noted earlier, the JSON format depends on ES and the schema (and possibly other aspects of how PonyMail uses ES). It's not guaranteed stable.

I think it would be quite a lot of work to define a format that is guaranteed not to change.

**Comment by dimidd** [comment]
I see, would it be possible somehow to infer it from the existing code
automagically? E.g. add an optional flag `--dump-json` to `import-mbox.py`.

2018-05-14 13:43 GMT+03:00 sebbASF <notifications@github.com>:

**Comment by sebbASF** [comment]
No; it does not work like that. There is not a simple JSON interface that can be exposed externally.

Whilst the JSON format can be captured for a given message by dumping the parameters to the ES methods, that is not the whole story. e.g. the message ID also needs to be generated; attachments scanned etc.

PonyMail requires the original source message so that will need to be part of your input data.
This can just be fed as input to the archiver. Or the source can be written to a file and parsed by import-mbox.
